app/src/main/java/org/simple/clinic/navigation/v2/fragments/BaseScreen.kt

```diff
@@ -76,7 +76,7 @@ abstract class BaseScreen<K : ScreenKey, B : ViewBinding, M : Parcelable, E, F,
 
     _viewModel = ViewModelProvider(viewModelStore, object : ViewModelProvider.Factory {
       private fun loop(viewEffectsConsumer: Consumer<V>) = RxMobius
-          .loop(createUpdate(), createEffectHandler())
+          .loop(createUpdate(), createEffectHandler(viewEffectsConsumer))
```

Comment:
As I understand it, we'll create a new kind of effect for representing view effects and instead of invoking UI actions in the current effect handler, we will pass the view effects to the view effect handler and invoke the UI actions from there instead?

Reply:
Yep, instead of calling `uiActions.methodName` in effect handler. We will just call `viewEffectsConsumer.accept` and send a view effect to Mobius loop. And that will be passed to view effects handler for triggering the UI actions.